### DISCUSSION TITLE
articles/5-tips-on-unit-testing/

### DISCUSSION THREAD

**Discussion by giscus** [piglei/piglei-giscus-comments#13] [General]
# articles/5-tips-on-unit-testing/

piglei 的 blog

https://www.piglei.com/articles/5-tips-on-unit-testing/

**Comment by Li-Major** [comment]
写得好好

**Comment by CapronerRevive** [comment]
非常好文字，使我重拾单测

> 但有了单元测试后，情况就完全不同了。每当你写出难以测试的代码时，单元测试总会无差别的大声告诉你：“你写的代码太烂了！”不留**不点情面**。

抓个虫，应该是**一点情面**

**Comment by KDL-in** [comment]
以前我比较同意一个观点：实际生产精力有限，应该把单测的精力都花在20%的核心代码上。工作中大量代码都比较简单，大概有20%是核心代码，我认为应该把精力花在这上边，而不是覆盖那些没啥意思的if-else。

现在有了ai不一样了，单测roi真的很高，主要是投入变得很低
